src/time_integration/paired_explicit_runge_kutta/methods_PERK3.jl

```diff
@@ -2,8 +2,117 @@
 # Since these FMAs can increase the performance of many numerical algorithms,
 # we need to opt-in explicitly.
 # See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+using NLsolve
 @muladd begin
 
+function F_own(aUnknown, NumStages, NumStageEvals, MonCoeffs)
+    cS2 = 1 # cS2 = c_ts(S-2)
+    c_ts = c_Own(cS2, NumStages) # ts = timestep
+
+    c_eq = zeros(NumStageEvals - 2) # Add equality constraint that cS2 is equal to 1
+    # Both terms should be present
+    for i in 1:(NumStageEvals - 4)
+        term1 = aUnknown[NumStageEvals - 1]
+        term2 = aUnknown[NumStageEvals]
+        for j in 1:i
+            term1 *= aUnknown[NumStageEvals - 1 - j]
+            term2 *= aUnknown[NumStageEvals - j]
+        end
+        term1 *= c_ts[NumStages - 2 - i] * 1/6
+        term2 *= c_ts[NumStages - 1 - i] * 4/6
+
+        c_eq[i] = MonCoeffs[i] - (term1 + term2)
+    end
+
+    # Highest coefficient: Only one term present
+    i = NumStageEvals - 3
+    term2 = aUnknown[NumStageEvals]
+    for j in 1:i
+        term2 *= aUnknown[NumStageEvals - j]
+    end
+    term2 *= c_ts[NumStages - 1 - i] * 4/6
+
+    c_eq[i] = MonCoeffs[i] - term2
+
+    c_eq[NumStageEvals - 2] = 1.0 - 4 * aUnknown[NumStageEvals] - aUnknown[NumStageEvals - 1]
+
+    return c_eq
+end
+
+function c_Own(cS2, NumStages)
+    c_ts = zeros(NumStages);
+
+    # Last timesteps as for SSPRK33
+    c_ts[NumStages]     = 0.5;
+    c_ts[NumStages - 1] = 1;
+
+    # Linear increasing timestep for remainder
+    for i = 2:NumStages-2
+        c_ts[i] = cS2 * (i-1)/(NumStages - 3);
+    end
+    return c_ts
+end
+
+function ComputePERK3_ButcherTableau(NumStages, NumStageEvals, semi::AbstractSemidiscretization)
+
+    #Initialize array of c
+    c_ts = c_Own(1.0, NumStages)
+
+    #Initialize the array of our solution
+    aUnknown = zeros(NumStageEvals)
+
+    #Case of e = 3
+    if NumStageEvals == 3
+        aUnknown = [0, c_ts[2], 0.25]
+
+    else
+        #Calculate MonCoeffs from polynomial Optimizer
+        ConsOrder = 3
+        dtMax = 1.0
+        dtEps = 1e-9
+        filter_thres = 1e-12
+
+        #Get EigVals
+        J = jacobian_ad_forward(semi)
+        EigVals = eigvals(J)
+        NumEigVals, EigVals = filter_Eigvals(EigVals, filter_thres)
+
+        MonCoeffs, _, _ = Bisection(ConsOrder, NumEigVals, NumStages, dtMax, dtEps, EigVals)
+        MonCoeffs = undo_normalization(ConsOrder, NumStages, MonCoeffs)
+
+        #Define the objective_function
+        function objective_function(x)
+            return F_own(x, NumStages, NumStageEvals, MonCoeffs)
+        end
+
+        #call nlsolver to solve until the answer is not NaN or negative values
+        is_sol_valid = false
+        while is_sol_valid  == false
+            #Initialize initial guess
+            x0 = 0.1 .* rand(NumStageEvals)
+            x0[1] = Base.big(0)
```

Comment:
```suggestion
            x0[1] = 0.0
```

We should not use `BigFloat` here, as this probably impairs performance.